src/h5web/vis-packs/nexus/utils.ts

```diff
@@ -26,6 +32,20 @@ export function getAttributeValue(
   return entity.attributes?.find((attr) => attr.name === attributeName)?.value;
 }
 
+export function getDatasetLabel(dataset: Dataset): string {
+  const longName = getAttributeValue(dataset, 'long_name');
+  if (longName && typeof longName === 'string') {
+    return longName;
+  }
+
+  const units = getAttributeValue(dataset, 'units');
+  if (units && typeof units === 'string') {
+    return `${dataset.name} (${units})`;
+  }
+
+  return dataset.name;
+}
+
```

Comment:
The diff is ugly because I've moved things around, but all I did was move the hooks here, rename them, and remove `useContext` and `prefetch` from each of them.